Yosemite/Yosemite/Model/Mocks/MockObjectGraph.swift

```diff
@@ -0,0 +1,481 @@
+import Foundation
+import Storage
+
+public protocol MockObjectGraph {
+    var userCredentials: Credentials { get }
+    var defaultAccount: Account { get }
+    var defaultSite: Site { get }
+    var defaultSiteAPI: SiteAPI { get }
+
+    var sites: [Site] { get }
+    var orders: [Order] { get }
+    var products: [Product] { get }
+    var reviews: [ProductReview] { get }
+
+    var thisYearOrderStats: OrderStatsV4 { get }
+    var thisYearVisitStats: SiteVisitStats { get }
+    var thisYearTopProducts: TopEarnerStats { get }
+
+    func accountWithId(id: Int64) -> Account
+    func accountSettingsWithUserId(userId: Int64) -> AccountSettings
+
+    func siteWithId(id: Int64) -> Site
+
+    var currentNotificationCount: Int { get }
+    var statsVersion: StatsVersion { get }
+
+    func statsV4ShouldBeAvailable(forSiteId: Int64) -> Bool
+}
+
+let mockResourceUrlHost = "http://localhost:\(UserDefaults.standard.integer(forKey: "mocks-port"))/"
+
+// MARK: Product Accessors
+extension MockObjectGraph {
+
+    func product(forSiteId siteId: Int64, productId: Int64) -> Product {
+        return products(forSiteId: siteId).first { $0.productID == productId }!
+    }
```

Comment:
Thinking out loud: seeing this force-unwrap I'm wondering: what's the real world app behavior when such productID doesn't exist, and wouldn't we want to be able to mock/test that "product not found in sideID" case in our tests? (iow return an optional here), like you do for `order(…)` and `review(…)` below? 🤔

Reply:
Yeah, good question – the product should be guaranteed to exist on the object graph here – we're just finding it in the list. Because `first` returns an optional, we force-unwrap it. If this ever fails there are bigger issues at play (ie – maybe we've loaded a production database in a test context?)